front/app/modules/commercial/insights/admin/components/TopBar/ProjectsDropdown.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from 'react';
+
+// styling
+import styled from 'styled-components';
+import { colors, fontSizes } from 'utils/styleUtils';
+
+// components
+import {
+  Dropdown,
+  DropdownListItem,
+  Icon,
+} from '@citizenlab/cl2-component-library';
+import Button from 'components/UI/Button';
+
+// intl
+import { injectIntl } from 'utils/cl-intl';
+import { InjectedIntlProps } from 'react-intl';
+import messages from './messages';
+
+// hooks
+import useProject from 'hooks/useProject';
+import useLocalize from 'hooks/useLocalize';
+
+// utils
+import { isNilOrError } from 'utils/helperUtils';
+
+const DropdownWrapper = styled.div`
+  color: ${colors.label};
+  font-size: ${fontSizes.small}px;
+
+  button {
+    font-weight: 500;
+    &:hover {
+      color: black;
+    }
+  }
+`;
+
+const StyledIcon = styled(Icon)`
+  width: 10px;
+  margin-left: 8px;
+`;
+
+interface ProjectTitleProps {
+  projectId: string;
+}
+
+const ProjectTitle = ({ projectId }: ProjectTitleProps) => {
+  const project = useProject({ projectId });
+  const localize = useLocalize();
+
+  if (isNilOrError(project)) return null;
+
+  const openProjectPage = () => {
+    window.open(`/projects/${project.attributes.slug}`, '_blank');
+  };
+
+  return (
+    <DropdownListItem onClick={openProjectPage}>
+      {localize(project.attributes.title_multiloc)}
+    </DropdownListItem>
+  );
+};
+
+interface Props {
+  projectIds: string[];
+}
+
+const ProjectsDropdown = ({
+  projectIds,
+  intl: { formatMessage },
+}: Props & InjectedIntlProps) => {
+  const [isDropdownOpened, setDropdownOpened] = useState(false);
+
+  const toggleDropdown = () => {
+    setDropdownOpened(!isDropdownOpened);
+  };
+
+  const closeDropdown = () => {
+    setDropdownOpened(false);
+  };
+
+  return (
+    <DropdownWrapper>
```

Comment:
We should really refactor this component, it feels we are reimplementing so much of the same logic every time we use it! Obviously out of scope for this PR, just a though 😅

Reply:
Totally agree. I was actually working on standardizing this for multiselect-ish components in the component library (see https://github.com/CitizenLabDotCo/cl2-component-library/blob/multi-select/src/components/MultiSelect/index.tsx), and if I had enough time I was hoping to extend it to the single select as well. But it got deprioritized because of my move to insights :/

Reply:
Too bad! I hope we can find the time to pick it up again soon.